tools/src/main/scala/caliban/tools/SchemaWriter.scala

```diff
@@ -260,11 +261,14 @@ object SchemaWriter {
     def writeInputValue(value: InputValueDefinition): String = {
       val GQLNewTypeInputDirective = writeGQLNewTypeDirective(value.directives)
 
-      val inputDef = resolveNewTypeInputDef(value).getOrElse(value)
+      val inputDef    = resolveNewTypeInputDef(value).getOrElse(value)
```

Comment:
This makes sense only if the field is nullable, right? I guess we shouldn't apply it for nullable fields?

Also what happens with the Option? Does it create `Either[Unit, Option[A]]`? If that's the case, it's not great if you want to use something like `TriState[A]`, I think that F should replace the `Option`.

Reply:
It makes sense for both nullable and not-nullable fields:
```
input UpdateValue {
  value: Int!
  description: String
}
```
`description` is nullable and a `null` value is expected, whereas 
`value` is not and we expect a number there. But the field still can me missing, in case the user doesn't want to change the value. Right or am I missing something?
If the above is correct, then `TriState[A]` won't work unfortunately as and other effect containing `Option`. Hence `F[Option[A]]` in case of nullable fields.

Reply:
> value is not and we expect a number there. But the field still can me missing, in case the user doesn't want to change the value. Right or am I missing something?

If the field is not nullable, it can't be null. The query validation will fail if you send null.

Reply:
Non-nullable field still can be missed in the input. Hence `F[A]`, where `A` may (or may not) be `Option[B]`.

Reply:
> Non-nullable field still can be missed in the input.

No, this is an invalid query then, unless it has a default value but then it will have that value.

Reply:
I've found it in the [GraphQL spec](https://spec.graphql.org/draft/#sec-Input-Objects.Input-Coercion). What should I do with the PR?

Reply:
It says the same thing I was saying (see highlights in bold):

> If no value is provided for a defined input object field and that field definition provides a default value, the default value should be used. If **no default value** is provided and the input object field’s type is **non-null**, **an error should be raised**. Otherwise, if the field is not required, then no entry is added to the coerced unordered map.

> If the value null was provided for an input object field, **and the field’s type is not a non-null type**, an entry in the coerced unordered map is given the value null. In other words, there is a semantic difference between the explicitly provided value null versus having not provided a value.